src/libLookup/Lookup.cpp

```diff
@@ -1590,6 +1590,99 @@ void Lookup::SendGetMicroBlockFromLookup(const vector<BlockHash>& mbHashes) {
   SendMessageToRandomLookupNode(msg);
 }
 
+bool Lookup::ProcessGetCosigsRewardsFromSeed(
+    [[gnu::unused]] const bytes& message, [[gnu::unused]] unsigned int offset,
+    [[gnu::unused]] const Peer& from) {
+  LOG_MARKER();
+
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Function not expected to be called from non-lookup node");
+    return false;
+  }
+
+  // verify if sender is from know DS Committee
+  uint128_t ipSenderAddr = from.m_ipAddress;
+  bool isDSNode = false;
+  for (const auto& node : *m_mediator.m_DSCommittee) {
+    if (ipSenderAddr == node.second.m_ipAddress) {
+      isDSNode = true;
+      break;
+    }
+  }
+  if (!isDSNode) {
+    LOG_GENERAL(
+        WARNING,
+        "Requesting IP : "
+            << from.GetPrintableIPAddress()
+            << " is not in Present DS Committee list. Ignore the request");
+    return false;
+  }
+
+  uint64_t blockNum;
+  uint32_t portNo = 0;
+  PubKey dsPubKey;
+  if (!Messenger::GetLookupGetCosigsRewardsFromSeed(message, offset, dsPubKey,
+                                                    blockNum, portNo)) {
```

Comment:
Since this only returns the cosigs for only one txblock but most of the scenario multiple txblocks are needed. Shall we make the output list of blockNum here for aggregating the queries? And perhaps a checking of blockNum is needed also  to avoid abusing the api for querying from outdated blocks.

Reply:
Intentionally not allowing asking for multiple tx blocks because after considering all cases, node should not be asking for cosigs for multiple txblocks from any single lookup which will also help to avoid overwhelming. 
Regarding blockNum check, yes it makes sense to check if older than curr ds epoch. will make the change.